src/decoder/mod.rs

```diff
@@ -710,6 +712,32 @@ impl<R: Read + Seek> Decoder<R> {
         }
     }
 
+    pub fn get_all_tags(&mut self) -> Vec<(Tag, ifd::Value)> {
+        let mut all_tags = Vec::new();
+
+        let dir = match &self.ifd {
+            Some(d) => d,
+            None => {
+                println!("No tags found");
+                return all_tags;
+            }
+        };
+        let keys = dir.keys().cloned().collect::<Vec<Tag>>();
+        // go through all tags
+        for key in keys {
+            let tag_value = match self.get_tag(key) {
+                Ok(val) => val,
+                _ => {
+                    println!("Could not unwrap {:?}", key);
+                    continue;
+                }
+            };
+            all_tags.push((key, tag_value));
+        }
+
+        all_tags
+    }
```

Comment:
This too appears to be on its own and unused. There's a lot of design space available for _just_ this method and I don't want to hold up the actual content of jpeg decoding over discussing this for too long.

Reply:
Yes this was useful for me. Do you want me to remove it?

Reply:
Yes please. Both printing and allocating the values is questionable for the public interface of the library. We can discuss this in a separate issue since I do see how it could be useful.

Reply:
Oh dang forgot the print. Will do.